### ISSUE TITLE
Not detecting files in subfolder to process

### ISSUE THREAD

**Issue by ray73864** [revenz/FileFlows#156]
I've set my library up to be for 'F:\TV Shows\Outlander' (Using the directory selector, not manually typing it in), I then force a refresh on that library, but it isn't detecting any of the files.

There are 75 files spread across 6 seasons, each season is in its own subfolder (The same as the rest of my Plex library)

```
F:\TV Shows\Outlander\Season 01
F:\TV Shows\Outlander\Season 02
etc...
```

When I check the Info log (downloading it, since it doesn't seem to want to display online), it says that it can't find any files.
I've checked every episode via Plex to see what the current encoding is, Season 01 is all in HEVC (Not converted via FileFlows though, I suspect Sonarr downloaded that 1 season as HEVC), however the remaining 5 seasons are all currently H264.

```
2022-06-28 10:52:04.220 [INFO] -> Triggering worker: LibraryWorker
2022-06-28 10:52:04.225 [INFO] -> WatchedLibrary: Library 'Video Library 2' marked for full scan
2022-06-28 10:52:04.227 [DBUG] -> Scan started on 'Video Library 2': F:\TV Shows\Outlander
2022-06-28 10:52:08.357 [INFO] -> Triggering worker: FlowRunnerMonitor
2022-06-28 10:52:08.359 [INFO] -> Triggering worker: FlowWorker
2022-06-28 10:52:08.372 [INFO] -> GetNextFile for ['FileFlowsServer'](bf47da28-051e-452e-ad21-c6a3f477fea9): NoFile
2022-06-28 10:52:08.374 [INFO] -> No file found to process, status from server: NoFile
```

Earlier I had it working on 'Superman & Lois', and it found files in that particular one, so I'm not entirely sure why it isn't finding any this time.

It's scanned the directory at least 4 times now, last time was 2mins ago.

**Comment by revenz** [comment]
Turn on this setting and try again.   
![image](https://user-images.githubusercontent.com/958400/176084401-79ec2449-5540-4feb-bb76-77a9ac01dd95.png)

This does excess logging, so turn it off afterwards, but can be helpful in these situations.

**Comment by ray73864** [comment]
All that did was give me 1 extra line in the logfile.

```
2022-06-28 11:28:31.333 [INFO] -> Triggering worker: LibraryWorker
2022-06-28 11:28:31.334 [INFO] -> WatchedLibrary: Library 'Video Library 2' marked for full scan
2022-06-28 11:28:31.336 [DBUG] -> Scan started on 'Video Library 2': F:\TV Shows\Outlander
2022-06-28 11:28:31.359 [DBUG] -> Files queued for 'Video Library 2': 0 / 0
```

**Comment by ray73864** [comment]
Thought maybe it was a problem with 1469 so I updated to 1559, now I have a new issue it looks like that is showing in the logfile, and the existing one with it not finding any files to process is still present.

```
2022-06-28 11:38:56.469 [INFO] -> Starting FileFlows 0.8.3.1559
2022-06-28 11:38:56.490 [DBUG] -> Arguments: No arguments
2022-06-28 11:38:56.507 [DBUG] -> ENV.ComSpec = C:\WINDOWS\system32\cmd.exe
2022-06-28 11:38:56.522 [DBUG] -> ENV.ProgramFiles = C:\Program Files
2022-06-28 11:38:56.538 [DBUG] -> ENV.CommonProgramFiles(x86) = C:\Program Files (x86)\Common Files
2022-06-28 11:38:56.553 [DBUG] -> ENV.windir = C:\WINDOWS
2022-06-28 11:38:56.570 [DBUG] -> ENV.USERPROFILE = C:\Users\Administrator
2022-06-28 11:38:56.585 [DBUG] -> ENV.USERNAME = Administrator
2022-06-28 11:38:56.601 [DBUG] -> ENV.FP_NO_HOST_CHECK = NO
2022-06-28 11:38:56.616 [DBUG] -> ENV.LOCALAPPDATA = C:\Users\Administrator\AppData\Local
2022-06-28 11:38:56.632 [DBUG] -> ENV.HOMEDRIVE = C:
2022-06-28 11:38:56.647 [DBUG] -> ENV.PATHEXT = .COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC
2022-06-28 11:38:56.664 [DBUG] -> ENV.OS = Windows_NT
2022-06-28 11:38:56.678 [DBUG] -> ENV.ProgramData = C:\ProgramData
2022-06-28 11:38:56.699 [DBUG] -> ENV.SESSIONNAME = RDP-Tcp#37
2022-06-28 11:38:56.709 [DBUG] -> ENV.PROCESSOR_LEVEL = 21
2022-06-28 11:38:56.725 [DBUG] -> ENV.SystemRoot = C:\WINDOWS
2022-06-28 11:38:56.740 [DBUG] -> ENV.LOGONSERVER = \\WIN-RTDLDA207TA
2022-06-28 11:38:56.756 [DBUG] -> ENV.HOMEPATH = \Users\Administrator
2022-06-28 11:38:56.772 [DBUG] -> ENV.ProgramW6432 = C:\Program Files
2022-06-28 11:38:56.787 [DBUG] -> ENV.CommonProgramFiles = C:\Program Files\Common Files
2022-06-28 11:38:56.803 [DBUG] -> ENV.PSModulePath = C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules\
2022-06-28 11:38:56.819 [DBUG] -> ENV.NUMBER_OF_PROCESSORS = 4
2022-06-28 11:38:56.834 [DBUG] -> ENV.ALLUSERSPROFILE = C:\ProgramData
2022-06-28 11:38:56.850 [DBUG] -> ENV.DriverData = C:\Windows\System32\Drivers\DriverData
2022-06-28 11:38:56.865 [DBUG] -> ENV.PROCESSOR_IDENTIFIER = AMD64 Family 21 Model 48 Stepping 1, AuthenticAMD
2022-06-28 11:38:56.881 [DBUG] -> ENV.COMPUTERNAME = WIN-RTDLDA207TA
2022-06-28 11:38:56.897 [DBUG] -> ENV.Path = C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\ProgramData\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\AMD\ATI.ACE\Core-Static;c:\winnmp\bin;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\dotnet\;;C:\WinNMP\bin;C:\Users\Administrator\AppData\Local\Microsoft\WindowsApps;C:\Users\Administrator\.dotnet\tools
2022-06-28 11:38:56.913 [DBUG] -> ENV.CLIENTNAME = DESKTOP-L80LRFO
2022-06-28 11:38:56.928 [DBUG] -> ENV.TMP = C:\Users\ADMINI~1\AppData\Local\Temp\2
2022-06-28 11:38:56.943 [DBUG] -> ENV.USERDOMAIN_ROAMINGPROFILE = WIN-RTDLDA207TA
2022-06-28 11:38:56.959 [DBUG] -> ENV.ProgramFiles(x86) = C:\Program Files (x86)
2022-06-28 11:38:56.975 [DBUG] -> ENV.CommonProgramW6432 = C:\Program Files\Common Files
2022-06-28 11:38:56.990 [DBUG] -> ENV.PUBLIC = C:\Users\Public
2022-06-28 11:38:57.006 [DBUG] -> ENV.USERDOMAIN = WIN-RTDLDA207TA
2022-06-28 11:38:57.023 [DBUG] -> ENV.TEMP = C:\Users\ADMINI~1\AppData\Local\Temp\2
2022-06-28 11:38:57.038 [DBUG] -> ENV.APPDATA = C:\Users\Administrator\AppData\Roaming
2022-06-28 11:38:57.054 [DBUG] -> ENV.PROCESSOR_REVISION = 3001
2022-06-28 11:38:57.069 [DBUG] -> ENV.SystemDrive = C:
2022-06-28 11:38:57.085 [DBUG] -> ENV.PROCESSOR_ARCHITECTURE = AMD64
2022-06-28 11:38:57.115 [INFO] -> ==================================================
2022-06-28 11:38:57.218 [INFO] -> Database initialized
2022-06-28 11:38:57.562 [INFO] -> Checking default Scripts
2022-06-28 11:38:57.719 [INFO] -> Database initialized
2022-06-28 11:38:58.161 [DBUG] -> Scanning for plugins
2022-06-28 11:38:58.163 [DBUG] -> Plugin path:C:\Users\Administrator\AppData\Roaming\FileFlows\Plugins
2022-06-28 11:38:58.255 [DBUG] -> Plugin file found: C:\Users\Administrator\AppData\Roaming\FileFlows\Plugins\BasicNodes.ffplugin
2022-06-28 11:38:58.273 [DBUG] -> Plugin.Name: Basic Nodes
2022-06-28 11:38:58.274 [DBUG] -> Plugin.PackageName: BasicNodes
2022-06-28 11:38:58.276 [DBUG] -> Plugin.Version: 0.8.3.313
2022-06-28 11:38:58.277 [DBUG] -> Plugin.Url: https://fileflows.com/
2022-06-28 11:38:58.277 [DBUG] -> Plugin.Authors: John Andrews
2022-06-28 11:38:58.278 [INFO] -> Updating plugin: Basic Nodes
2022-06-28 11:38:58.306 [DBUG] -> Plugin file found: C:\Users\Administrator\AppData\Roaming\FileFlows\Plugins\ImageNodes.ffplugin
2022-06-28 11:38:58.307 [DBUG] -> Plugin.Name: Image Nodes
2022-06-28 11:38:58.309 [DBUG] -> Plugin.PackageName: ImageNodes
2022-06-28 11:38:58.311 [DBUG] -> Plugin.Version: 0.8.3.313
2022-06-28 11:38:58.312 [DBUG] -> Plugin.Url: https://fileflows.com/
2022-06-28 11:38:58.313 [DBUG] -> Plugin.Authors: John Andrews
2022-06-28 11:38:58.314 [INFO] -> Updating plugin: Image Nodes
2022-06-28 11:38:58.328 [DBUG] -> Plugin file found: C:\Users\Administrator\AppData\Roaming\FileFlows\Plugins\MusicNodes.ffplugin
2022-06-28 11:38:58.329 [DBUG] -> Plugin.Name: Music Nodes
2022-06-28 11:38:58.330 [DBUG] -> Plugin.PackageName: MusicNodes
2022-06-28 11:38:58.332 [DBUG] -> Plugin.Version: 0.8.3.313
2022-06-28 11:38:58.333 [DBUG] -> Plugin.Url: https://fileflows.com/
2022-06-28 11:38:58.334 [DBUG] -> Plugin.Authors: John Andrews
2022-06-28 11:38:58.335 [INFO] -> Updating plugin: Music Nodes
2022-06-28 11:38:58.350 [DBUG] -> Plugin file found: C:\Users\Administrator\AppData\Roaming\FileFlows\Plugins\VideoLegacyNodes.ffplugin
2022-06-28 11:38:58.353 [DBUG] -> Plugin.Name: Video Legacy Nodes
2022-06-28 11:38:58.356 [DBUG] -> Plugin.PackageName: VideoLegacyNodes
2022-06-28 11:38:58.357 [DBUG] -> Plugin.Version: 0.8.3.313
2022-06-28 11:38:58.358 [DBUG] -> Plugin.Url: https://fileflows.com/
2022-06-28 11:38:58.359 [DBUG] -> Plugin.Authors: John Andrews
2022-06-28 11:38:58.359 [INFO] -> Updating plugin: Video Legacy Nodes
2022-06-28 11:38:58.377 [DBUG] -> Plugin file found: C:\Users\Administrator\AppData\Roaming\FileFlows\Plugins\VideoNodes.ffplugin
2022-06-28 11:38:58.392 [DBUG] -> Plugin.Name: Video Nodes
2022-06-28 11:38:58.395 [DBUG] -> Plugin.PackageName: VideoNodes
2022-06-28 11:38:58.396 [DBUG] -> Plugin.Version: 0.8.3.313
2022-06-28 11:38:58.397 [DBUG] -> Plugin.Url: https://fileflows.com/
2022-06-28 11:38:58.398 [DBUG] -> Plugin.Authors: John Andrews
2022-06-28 11:38:58.398 [INFO] -> Updating plugin: Video Nodes
2022-06-28 11:38:58.432 [INFO] -> Finished scanning for plugins
2022-06-28 11:38:58.547 [DBUG] -> Language keys found: 356
2022-06-28 11:39:03.201 [INFO] -> Triggering worker: LibraryWorker
2022-06-28 11:39:03.203 [INFO] -> Triggering worker: LogFileCleaner
2022-06-28 11:39:03.204 [INFO] -> Triggering worker: PluginUpdaterWorker
2022-06-28 11:39:03.208 [INFO] -> Triggering worker: TelemetryReporter
2022-06-28 11:39:03.211 [INFO] -> Plugin Updater started
2022-06-28 11:39:03.225 [INFO] -> Triggering worker: TempFileCleaner
2022-06-28 11:39:03.237 [INFO] -> Starting worker: LicenseValidatorWorker
2022-06-28 11:39:03.252 [INFO] -> Making request[GET]: https://fileflows.com/api/plugin?rand=133008611432176645
2022-06-28 11:39:03.253 [INFO] -> Making request[GET]: http://localhost:5000/api/settings
2022-06-28 11:39:03.256 [INFO] -> Starting worker: LicenseValidatorWorker
2022-06-28 11:39:03.257 [INFO] -> Starting worker: LibraryWorker
2022-06-28 11:39:03.258 [INFO] -> Starting worker: LibraryWorker
2022-06-28 11:39:03.323 [INFO] -> Starting worker: LogFileCleaner
2022-06-28 11:39:03.325 [INFO] -> Starting worker: LogFileCleaner
2022-06-28 11:39:03.326 [INFO] -> Starting worker: DbLogPruner
2022-06-28 11:39:03.327 [INFO] -> Starting worker: DbLogPruner
2022-06-28 11:39:03.356 [INFO] -> Starting worker: FlowWorker
2022-06-28 11:39:03.360 [INFO] -> Starting worker: FlowWorker
2022-06-28 11:39:03.375 [INFO] -> Starting worker: PluginUpdaterWorker
2022-06-28 11:39:03.387 [INFO] -> Starting worker: PluginUpdaterWorker
2022-06-28 11:39:03.388 [INFO] -> Starting worker: LibraryFileLogPruner
2022-06-28 11:39:03.389 [INFO] -> Starting worker: LibraryFileLogPruner
2022-06-28 11:39:03.390 [INFO] -> Starting worker: LogConverter
2022-06-28 11:39:03.391 [INFO] -> Starting worker: LogConverter
2022-06-28 11:39:03.392 [INFO] -> Starting worker: TelemetryReporter
2022-06-28 11:39:03.393 [INFO] -> Starting worker: TelemetryReporter
2022-06-28 11:39:03.394 [INFO] -> Starting worker: ServerUpdater
2022-06-28 11:39:03.395 [INFO] -> Starting worker: ServerUpdater
2022-06-28 11:39:03.396 [INFO] -> Starting worker: TempFileCleaner
2022-06-28 11:39:03.397 [INFO] -> Starting worker: TempFileCleaner
2022-06-28 11:39:03.397 [INFO] -> Starting worker: FlowRunnerMonitor
2022-06-28 11:39:03.398 [INFO] -> Starting worker: FlowRunnerMonitor
2022-06-28 11:39:03.400 [INFO] -> Starting worker: SystemMonitor
2022-06-28 11:39:03.401 [INFO] -> Starting worker: SystemMonitor
2022-06-28 11:39:03.403 [INFO] -> Processing Timer Queue: Video Library
2022-06-28 11:39:03.404 [INFO] -> Processing Timer Queue: Video Library 2
2022-06-28 11:39:03.406 [DBUG] -> Scan started on 'Video Library 2': F:\TV Shows\Outlander
2022-06-28 11:39:03.413 [INFO] -> About to clean temporary directory: F:\temp
2022-06-28 11:39:06.551 [INFO] -> Deleted log file: FileFlows-Jun28-01
2022-06-28 11:39:06.828 [INFO] -> Plugin Updater finished
2022-06-28 11:39:09.127 [DBUG] -> Files queued for 'Video Library 2': 0 / 0
2022-06-28 11:39:09.149 [INFO] -> Making request[POST]: https://fileflows.com/api/telemetry
2022-06-28 11:39:10.578 [INFO] -> ServerUpdater: Current version '0.8.3.1559' newer or same as online version '0.8.3.1559'
2022-06-28 11:39:13.235 [INFO] -> Triggering worker: SystemMonitor
2022-06-28 11:39:13.300 [INFO] -> Triggering worker: LibraryWorker
2022-06-28 11:39:13.323 [INFO] -> Triggering worker: FlowWorker
2022-06-28 11:39:13.324 [INFO] -> Clearing workers
2022-06-28 11:39:13.326 [INFO] -> Triggering worker: FlowRunnerMonitor
2022-06-28 11:39:13.328 [INFO] -> Making request[GET]: http://localhost:5000/api/node/by-address/INTERNAL_NODE
2022-06-28 11:39:13.438 [INFO] -> GetNextFile for ['FileFlowsServer'](bf47da28-051e-452e-ad21-c6a3f477fea9): NoFile
2022-06-28 11:39:13.441 [INFO] -> No file found to process, status from server: NoFile
2022-06-28 11:39:21.273 [ERRR] -> Error in worker 'SystemMonitor': Object reference not set to an instance of an object.
   at FileFlows.Server.Database.Managers.DbManager.GetOpenDbConnections() in /src/Server/Database/Managers/DbManager.cs:line 133
   at FileFlows.Server.Workers.SystemMonitor.Execute() in /src/Server/Workers/SystemMonitor.cs:line 59
   at FileFlows.ServerShared.Workers.Worker.<Trigger>b__17_0() in /src/ServerShared/Workers/Worker.cs:line 115
2022-06-28 11:39:23.264 [INFO] -> Triggering worker: SystemMonitor
2022-06-28 11:39:23.288 [INFO] -> Triggering worker: FlowRunnerMonitor
2022-06-28 11:39:23.290 [INFO] -> Triggering worker: FlowWorker
2022-06-28 11:39:23.291 [INFO] -> Making request[GET]: http://localhost:5000/api/node/by-address/INTERNAL_NODE
2022-06-28 11:39:23.292 [INFO] -> GetNextFile for ['FileFlowsServer'](bf47da28-051e-452e-ad21-c6a3f477fea9): NoFile
2022-06-28 11:39:23.294 [INFO] -> No file found to process, status from server: NoFile
```

**Comment by revenz** [comment]
```
2022-06-28 11:39:21.273 [ERRR] -> Error in worker 'SystemMonitor': Object reference not set to an instance of an object.
   at FileFlows.Server.Database.Managers.DbManager.GetOpenDbConnections() in /src/Server/Database/Managers/DbManager.cs:line 133
   at FileFlows.Server.Workers.SystemMonitor.Execute() in /src/Server/Workers/SystemMonitor.cs:line 59
   at FileFlows.ServerShared.Workers.Worker.<Trigger>b__17_0() in /src/ServerShared/Workers/Worker.cs:line 115
```
doest that appear over and over or just once?

also can you turn on that setting I asked, on the latest version, hit rescan and give it a minute then upload that full log?

**Comment by ray73864** [comment]
Seems to happen every time it goes to scan the library, didn't happen in 1469, only appears to be a new thing in 1559.

I shutdown FileFlows, deleted the logfile, started it up again, had it scan the library, left it for a minute or 2, and have now attached the full logfile (This is with that setting turned on).

[FileFlows-Jun28-01.log](https://github.com/revenz/FileFlows/files/8997570/FileFlows-Jun28-01.log)

**Comment by revenz** [comment]
k, that error is already fixed, why I didnt see it.  So just ignore that.

Regarding why it cant see files in the folder.

There are 3 reasons why it wont see them
1. Files are already in the database.   Check your sqlite db file, or post here and I'll check
2. The file is unmatched by the filter or excluded via the exclude filter on the library.   
3. the file ends with _

**Comment by ray73864** [comment]
Interestingly, if I browse the sqlite db using "DB Browser for SQLite", I can see all 6 seasons of Outlander in it, however if I go to Files -> Processed (Which is currently at 2230), set the list to show 5000 on it, Outlander doesn't show at all, the filter can't even find it.

It looks like Outlander was done 7 days ago.

Since the processed section doesn't show them, what's the best way to force a reprocess? I suspect I'll have to delete them from the sqlite db directly?

**Comment by revenz** [comment]
If you click "Search" 
![image](https://user-images.githubusercontent.com/958400/176093995-19dc2b59-4eb1-44bb-8056-6811dc775ffb.png)

On the file page.,   you should be able to find them in there and delete them.

The "Files" only shows files inside a library, so if you deleted the library then readded it, the library UID will be different and the old files will be hidden.  In later versions of FF, it defaults to removing files when a library is deleted to avoid this.   
That does effect your overall statistics though as those are also deleted.

**Comment by ray73864** [comment]
Ahh ok.

Interestingly, when I go to that search and look for 'Outlander', it comes up with them all, but shows them as 'Unprocessed'.

In my mind, if the database says they are unprocessed, then it should try and process them, I dunno.

**Comment by revenz** [comment]
Agreed, its confusing.  I'm calling it a bug.   Just happens if you delete the library.   but yes, I will fix it.... somehow...

**Comment by revenz** [comment]
![image](https://user-images.githubusercontent.com/958400/176098326-49a9647b-5830-402f-a8bd-ddbf65fd47c3.png)

fixed in 0.8.4 new file status which show these files

**Comment by ray73864** [comment]
Awesome, thanks for that.

I'm still not 100% sure why Outlander wasn't showing, it can't be because of the library itself, as I've been using this library for a while.

I've changed the directory it has pointing at quite often, and when I go to the processed files, it shows 'Pokemon' there, which I did 2 weeks ago, and shows other directories I did going back to when I first started using FileFlows, even shows Superman & Lois which I processed earlier today, yet didn't show Outlander from a week ago.

Keeping in mind, I'm not deleting the library, I'm just going into the existing one, changing the path it points to, then rescanning it.
